src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/IncentiveStoreService.kt

```diff
@@ -38,4 +39,36 @@ class IncentiveStoreService(
     log.debug("${savedReviews.count()} records saved")
     nextReviewDateUpdaterService.update(remainingBookingId)
   }
+
+  suspend fun deleteIncentiveRecord(incentiveRecord: PrisonerIepLevel) {
+    prisonerIepLevelRepository.delete(incentiveRecord)
+    nextReviewDateUpdaterService.update(incentiveRecord.bookingId)
+
+    // If the deleted record had `current=true`, the latest IEP review becomes current
+    if (incentiveRecord.current) {
+      // The deleted record was current, set new current to the latest IEP review
+      prisonerIepLevelRepository.findFirstByBookingIdOrderByReviewTimeDesc(incentiveRecord.bookingId)?.run {
+        prisonerIepLevelRepository.save(this.copy(current = true))
+      }
+    }
+  }
+
+  suspend fun updateIncentiveRecord(
+    update: IncentiveRecordUpdate,
+    incentiveRecord: PrisonerIepLevel,
+  ): PrisonerIepLevel {
+    update.current?.let {
```

Comment:
i think we need this to be
```kotlin
if (update?.current) {
```
otherwise i'm pretty certain the block runs on any non-null value of current (including false)

Reply:
I think you're right, and I think we had a similar problem before, I need to check tho.